### PR TITLE
Azure skip availability set creation

### PR DESCRIPTION
**Pull request by florinpapa**
This change adds the `--azure_availability_set` flag, which controls whether an availability set is created. By default, the flag is set to `True`, but it can be manually set to `False` in order to skip availability set creation, if needed.

Fixes https://github.com/GoogleCloudPlatform/PerfKitBenchmarker/issues/1863.

Tested with mongodb_ycsb, as reported in the issue:
```
./pkb.py --cloud=Azure --benchmarks=mongodb_ycsb --benchmark_config_file=ycsb_config.yml --azure_availability_set=False

cat ycsb_config.yml
server: &server
    Azure:
        machine_type: Standard_L32s
        zone: eastus2
disk: &disk
    Azure:
        disk_type: local
        mount_point: /scratch
client: &client
    Azure:
        machine_type: Standard_F72s_v2
        zone: eastus2
mongodb_ycsb:
    vm_groups:
        workers:
            os_type: ubuntu1604
            vm_spec: *server
            disk_spec: *disk
            vm_count: 1
        clients:
            os_type: ubuntu1604
            vm_spec: *client
```

### CODE REVIEW

**Comment by florinpapa** [comment]
Can I please get feedback on this submission?

File: perfkitbenchmarker/providers/azure/azure_network.py [hunk 1]
```diff
@@ -457,7 +457,8 @@ def __init__(self, spec):
     self.location = util.GetLocationFromZone(self.zone)
     # With dedicated hosting and/or an availability zone, an availability set
     # cannot be created
-    if FLAGS.dedicated_hosts or util.GetAvailabilityZoneFromZone(self.zone):
+    if FLAGS.dedicated_hosts or util.GetAvailabilityZoneFromZone(self.zone) or \
```
Comment (shyamsabhaya) [inline perfkitbenchmarker/providers/azure/azure_network.py]: Back slash is not glint compliant. Recommend putting conditions in parenthesis instead.

**Comment by shyamsabhaya** [comment]
This change looks good. Be aware that we are currently working on a flag that will disable azure availability set. This upcoming flag will be `--placement_group`, setting this flag to `'none'` will not create a Azure availability set. The `--placement_group flag` will make this Azure specific `--azure_availability_set` flag obsolete, so we will remove it when functionality for `--placement_group` for Azure is added.

**Comment by florinpapa** [comment]
@shyamsabhaya thank you the feedback and for letting know regarding the upcoming `--placement-group` flag and its impact on this submission. I have added the necessary changes, please let me know what you think.

**Pull request merged**